backend/src/controllers/review.controller.ts

```diff
@@ -29,6 +41,34 @@ class ReviewController extends BaseController<IReview> {
     return res.status(httpStatus.OK).send(analysis);
   }
 
+  async generateResponseForReview(req: AuthRequest<{}, IReviewReply, IGenerateReviewReply>, res: Response) {
+    // TODO: implement this function with calls to OpenAPI
+    const { reviewText, prompt, previousReplies } = req.body;
+    await new Promise((resolve) => setTimeout(resolve, 2000));
+    res.status(httpStatus.OK).send({
+      text: `This should return a generated response for the review: ${reviewText} with ${previousReplies.length} previous replies and the prompt "${prompt}"`,
+    });
+  }
+
+  async getPaginated(req: AuthRequest<{}, {}, {}, IGetReviewsParams>, res: Response<IGetAllReviewsResponse>) {
+    const { limit, page, before, search } = req.query;
+
+    try {
+      const reviews = (await this.model
+        .find({ businessId: req.user!.businessId, date: { $lt: before } })
+        .limit(Number(limit))
+        .skip(Number(page - 1) * Number(limit))) as IReview[];
+
+      return res.status(httpStatus.OK).send({
+        currentPage: Number(page),
+        totalReviews: await this.model.countDocuments({ businessId: req.user!.businessId }),
```

Comment:
maybe we should set an index on the `businessId` property in the collection

Reply:
what do you mean?

Reply:
In the model file we can define DB indexes. This will make queries like this much faster. see [here](https://mongoosejs.com/docs/guide.html#indexes)